### PR TITLE
[SYCL] Reinitialize moved containers

### PR DESCRIPTION
**Pull request by s-kanaev**
Moved objects are in "valid but unspecified state". Using moved objects leads to undefined behaviour. It's okay to only clear moved containers if `clear` method doesn't imply any preconditions. Hence we will reinitialize the containers just to be sure.

### CODE REVIEW

File: sycl/source/detail/queue_impl.cpp [hunk 1]
```diff
@@ -217,6 +217,11 @@ void queue_impl::wait(const detail::code_location &CodeLoc) {
     std::lock_guard<mutex_class> Lock(MMutex);
     Events = std::move(MEventsWeak);
     USMEvents = std::move(MEventsShared);
+
+    // moved objects are in "valid but unspecified state". It's okay to only
+    // clear the vector. Though, we'll recostruct it hust to be sure.
```
Comment (cperkinsintel) [inline sycl/source/detail/queue_impl.cpp:222]: Good catch! LGTM.

I'm bad at wording comments, but maybe  "moved objects are in 'valid but unspecified state'.  For safety, we reconstruct empty vectors."

**Review by cperkinsintel — DISMISSED** [review]

File: sycl/source/detail/queue_impl.cpp [hunk 2]
```diff
@@ -217,6 +217,11 @@ void queue_impl::wait(const detail::code_location &CodeLoc) {
     std::lock_guard<mutex_class> Lock(MMutex);
     Events = std::move(MEventsWeak);
     USMEvents = std::move(MEventsShared);
```
Comment (romanovvlad) [inline sycl/source/detail/queue_impl.cpp:219]: Let's use `std::vector::swap` or `Events = vector_class<std::weak_ptr<event_impl>>(std::move(MEventsWeak));`
In move c'tor moved from std::vector is guaranteed to by empty.

Comment (s-kanaev) [reply sycl/source/detail/queue_impl.cpp:219]: Using `std::vector::swap` sounds good.

File: sycl/source/detail/queue_impl.hpp [hunk 3]
```diff
@@ -227,6 +227,8 @@ class queue_impl {
     {
       std::lock_guard<mutex_class> Lock(MMutex);
       Exceptions = std::move(MExceptions);
+
+      MExceptions = exception_list{};
```
Comment (romanovvlad) [inline sycl/source/detail/queue_impl.hpp:231]: Could you please use swap here as well?

Comment (s-kanaev) [reply sycl/source/detail/queue_impl.hpp]: Done

**Review by againull — APPROVED** [review]

**Review by cperkinsintel — APPROVED** [review]

**Pull request merged**